R/trailing_assignment_linter.R

```diff
@@ -0,0 +1,44 @@
+#' Trailing assignment linter
+#'
+#' Check that there are no assignment operators at the end of source lines.
+#'
+#' @param allow_comments Suppress lints for commented out lines
+#'
+#' @evalRd rd_tags("trailing_whitespace_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+trailing_assignment_linter <- function(allow_comments = FALSE) {
+  Linter(function(source_expression) {
+    if (!is_lint_level(source_expression, "file")) {
+      return(list())
+    }
+
+    res <- re_matches(
+      source_expression$file_lines,
+      rex(capture(or("<<-", "<-", "=", "->", "-->")), any_spaces, end)
+    )
+    bad_lines <- which(!vapply(res[, 1], is.na, logical(1), USE.NAMES = FALSE))
```

Comment:
This is creating a `implicit_integer_linter()` lint.

```suggestion
    bad_lines <- which(!vapply(res[, 1], is.na, 1L, USE.NAMES = FALSE))
```